Native plugin fails if no tests are run in a module

The native-maven-plugin will fail if no tests are run for some module after running `mvn test -Pnative`. It fails with the error "Test configuration file wasn't found. Make sure that test execution wasn't skipped." (No test config file will be written if no tests are run)

Error:
```
[ERROR] Test configuration file wasn't found. Make sure that test execution wasn't skipped.
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for native-junit-experiment 1.0-SNAPSHOT:
[INFO] 
[INFO] native-junit-experiment ............................ SUCCESS [  0.051 s]
[INFO] empty-test-module .................................. FAILURE [  0.544 s]
[INFO] simple-module ...................................... SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  0.860 s
[INFO] Finished at: 2022-01-06T10:09:52-05:00
```

Repro: https://github.com/dzou/native-plugin-empty-tests

I would expect the tests to just pass if no tests are run. This is important for multi-module projects because we use a test-filter which only runs native-image integration tests, and in multi-module projects, not all modules have native-image tests. It would then be useful in these cases to have the tests to pass if there's no tests that are run for some module.

Applicable to Gradle as well:
```
> Task :nativeTestCompile FAILED

FAILURE: Build failed with an exception.

* What went wrong:
A problem was found with the configuration of task ':nativeTestCompile' (type 'BuildNativeImageTask').
  - In plugin 'org.graalvm.buildtools.gradle.NativeImagePlugin$Inject' type 'org.graalvm.buildtools.gradle.tasks.BuildNativeImageTask' property 'testListDirectory' specifies directory 'native-plugin-empty-tests/empty-test-module/build/test-results/test/testlist' which doesn't exist.
    
    Reason: An input file was expected to be present but it doesn't exist.
    
    Possible solutions:
      1. Make sure the directory exists before the task is called.
      2. Make sure that the task which produces the directory is declared as an input.
    
    Please refer to https://docs.gradle.org/7.4/userguide/validation_problems.html#input_file_does_not_exist for more details about this problem.
```

--- a/native-gradle-plugin/src/main/java/org/graalvm/buildtools/gradle/NativeImagePlugin.java
+++ b/native-gradle-plugin/src/main/java/org/graalvm/buildtools/gradle/NativeImagePlugin.java
@@ -410,7 +410,7 @@ public class NativeImagePlugin implements Plugin<Project> {
         configurationRepository.getEnabled().convention(false);
         configurationRepository.getUri().convention(configurationRepository.getVersion().map(v -> {
             try {
-                return new URI("https://github.com/graalvm/jvm-reachability-metadata/releases/download/" + v + "/jvm-reachability-metadata-" + v + ".zip");
+                return new URI("https://github.com/graalvm/graalvm-reachability-metadata/releases/download/" + v + "/graalvm-reachability-metadata-" + v + ".zip");
             } catch (URISyntaxException e) {
                 return null;
             }
@@ -471,7 +471,7 @@ public class NativeImagePlugin implements Plugin<Project> {
         injectTestPluginDependencies(project, graalExtension.getTestSupport());
 
         TaskProvider<BuildNativeImageTask> testImageBuilder = tasks.named(deriveTaskName(name, "native", "Compile"), BuildNativeImageTask.class, task -> {
-            task.setOnlyIf(t -> graalExtension.getTestSupport().get());
+            task.setOnlyIf(t -> graalExtension.getTestSupport().get() && testListDirectory.getAsFile().get().exists());
             task.getTestListDirectory().set(testListDirectory);
             testTask.get();
             ConfigurableFileCollection testList = project.getObjects().fileCollection();
@@ -488,7 +488,7 @@ public class NativeImagePlugin implements Plugin<Project> {
 
         tasks.named(isPrimaryTest ? NATIVE_TEST_TASK_NAME : "native" + capitalize(name), NativeRunTask.class, task -> {
             task.setGroup(LifecycleBasePlugin.VERIFICATION_GROUP);
-            task.setOnlyIf(t -> graalExtension.getTestSupport().get());
+            task.setOnlyIf(t -> graalExtension.getTestSupport().get() && testListDirectory.getAsFile().get().exists());
         });
     }
 

--- a/native-maven-plugin/src/main/java/org/graalvm/buildtools/maven/NativeTestMojo.java
+++ b/native-maven-plugin/src/main/java/org/graalvm/buildtools/maven/NativeTestMojo.java
@@ -60,7 +60,6 @@ import java.io.File;
 import java.io.IOException;
 import java.io.UncheckedIOException;
 import java.net.URISyntaxException;
-import java.nio.file.DirectoryStream;
 import java.nio.file.Files;
 import java.nio.file.Path;
 import java.nio.file.Paths;
@@ -109,6 +108,7 @@ public class NativeTestMojo extends AbstractNativeMojo {
         }
         configureEnvironment();
         if (!hasTests()) {
+            logger.info("Skipped native-image tests since there are no test classes.");
             return;
         }
 
@@ -169,8 +169,8 @@ public class NativeTestMojo extends AbstractNativeMojo {
     private boolean hasTests() {
         Path testOutputPath = Paths.get(project.getBuild().getTestOutputDirectory());
         if (Files.exists(testOutputPath) && Files.isDirectory(testOutputPath)) {
-            try (DirectoryStream<Path> directory = Files.newDirectoryStream(testOutputPath)) {
-                return directory.iterator().hasNext();
+            try (Stream<Path> testClasses = Files.walk(testOutputPath)) {
+                return testClasses.anyMatch(p -> p.getFileName().toString().endsWith(".class"));
             } catch (IOException ex) {
                 throw new IllegalStateException(ex);
             }

